src/resource_manager/gateway.py

```diff
@@ -62,16 +63,30 @@ def _label_selector(self) -> str:
         return ",".join(f"{k}={v}" for k, v in self._labels.items())
 
     @map_k8s_auth_exception
-    def _gen_resource(self, definition: GatewayResourceDefinition, config: CharmConfig) -> dict:
+    def _gen_resource(self, definition: GatewayResourceDefinition, *args: typing.Any) -> dict:
```

Comment:
Nitpick:
I'm wondering if we could just method overloading for these rather than have them as typing.Any? That would help reduce the dynamic input checking.

Reply:
https://peps.python.org/pep-3124/#overloading-generic-functions